### PR TITLE
reset all metrics at each call, use integer instead of one-hot as target

### PR DESCRIPTION
**Pull request by smilesun**

### CODE REVIEW

**Comment by xinyuejohn** [comment]
@smilesun Good job! I will check this very soon!

**Comment by smilesun** [comment]
cool, i will merge it after your review !

File: domainlab/utils/perf_metrics.py [hunk 1]
```diff
@@ -44,13 +51,14 @@ def cal_metrics(self, model, loader_te, device, max_batches=None):
                 x_s, y_s = x_s.to(device), y_s.to(device)
                 pred_label, prob, _, *_ = model_local.infer_y_vpicn(x_s)
```
Comment (xinyuejohn) [inline domainlab/utils/perf_metrics.py:52]: @smilesun Could you tell me the returned shape of pred_label? 

Because when inputs are labels, inputs to the classification metrics need to be both in the shape of (N,).

File: domainlab/utils/perf_metrics.py [hunk 2]
```diff
@@ -44,13 +51,14 @@ def cal_metrics(self, model, loader_te, device, max_batches=None):
...
                 _, target_label = torch.max(y_s, 1)
```
Comment (xinyuejohn) [inline domainlab/utils/perf_metrics.py:53]: Here, the target_label is in shape of (N,).

File: domainlab/utils/perf_metrics.py (hunk 1 above)
Comment (smilesun) [reply domainlab/utils/perf_metrics.py:52]: Multi-class with logits or probabilities | (N, C) | float | (N,) | int
-- | -- | -- | -- | --

Comment (smilesun) [reply domainlab/utils/perf_metrics.py:52]: i will check it and let you know

Comment (smilesun) [reply domainlab/utils/perf_metrics.py:52]: pred_label.shape
torch.Size([100, 10])

Comment (smilesun) [reply domainlab/utils/perf_metrics.py:52]: that means if we change everything to probability, it should be fine right?

Comment (xinyuejohn) [reply domainlab/utils/perf_metrics.py]: Yes. But if we are using probability, do we need to pass a threshold? By default, the  classification metrics function uses threshold=0.5. I don't know if the model's threshold is also the same.

Comment (smilesun) [reply domainlab/utils/perf_metrics.py]: I don't think we need to pass threshold, since only the largetst logits will be taken as the predicted label. 
threshold only make since for binary classification AUC. However, I am not sure exaclty how multi-class AUC works

**Review by xinyuejohn — APPROVED** [review]
